boot/freeldr/freeldr/lib/fs/btrfs.c

```diff
@@ -699,18 +699,32 @@ static u64 btrfs_read_extent_reg(struct btrfs_path *path, struct btrfs_file_exte
         physical += extent->offset + offset;
         if (physical & (512 - 1))
         {
+            u32 shift;
+
             /* If somebody tried to do unaligned access */
-            physical -= offset;
-            temp_out = FrLdrTempAlloc(size + offset, TAG_BTRFS_FILE);
 
-            if (!disk_read(physical, temp_out, size + offset))
+            temp_out = FrLdrTempAlloc(512, TAG_BTRFS_FILE);
```

Comment:
Would `char temp_out[512]` on stack work?

Reply:
The stack is small here